rfc-128-continuous-deployment.md

```diff
@@ -0,0 +1,248 @@
+
+> **We will disable CD for Publishing API**. This was only enabled on a trial basis, and requires non-trivial work to bring it up to the new standard. We should avoid setting a false precedent by leaving CD enabled.
+
+The following steps must be taken as part of enabling automatic deployments:
+
+- Restrict merge access to "GOV.UK Production" [[example](https://github.com/alphagov/govuk-saas-config/pull/55)].
+
+- Tag [relevant Smoke tests](https://github.com/alphagov/smokey/blob/5dc36fc78c12838ec1c8070d6fbc535ae5433b89/docs/writing-tests.md#filtering-by-app) to run for the app [[example](https://github.com/alphagov/smokey/pull/708/commits/91a5ce94b7b96d09f5db87e63fbc3e400788de00)].
+
+- Add a note in the [Release](https://github.com/alphagov/release) app to warn about the change.
+
+- Add a PR template to warn about the change [[example](https://github.com/alphagov/smart-answers/blob/3509ba3e68c62a206874c96111628dd65adae689/.github/pull_request_template.md)].
+
+  - Backfill the PR template as a comment on existing PRs.
+
+- Enable automatic deployments in a Puppet PR [[example](https://github.com/alphagov/govuk-puppet/pull/10590)].
+
+  - Comment on how the app meets the criteria of this RFC.
+
+> **We can be flexible with coverage**. We can defer adding internal repo tests if the code has no impact on the running app e.g. rake tasks [[1](https://github.com/alphagov/email-alert-api/blob/59fc71a58317ef2998f2c0ef102020da3ca9df96/lib/tasks/support.rake)]. But we should not defer removing any unused code.
+
+### Delete [publishing-e2e-tests](https://github.com/alphagov/publishing-e2e-tests)
+
+In establishing the safety criteria for automatic deployments, we had to consider these tests. As explained in the rejected approaches, we chose to avoid them entirely. This brings their existence into question.
+
+We believe these tests add little or no value to GOV.UK. They significantly reduce our agility by adding 5+ minutes onto every PR build, to the extent that we sometimes remove them as a required check. They have very significant maintenance issues and only ever break for reasons unrelated to the features they are testing. Despite these signficant problems, we invest very little effort in improving them, and actively avoid extending them.
+
+We will disable and delete these tests, along with all of their associated infrastructure [[1](https://github.com/alphagov/email-alert-api/blob/0b87e62288ddb1653451f84e1f36e17ce4e8e9dc/Dockerfile)] [[2](https://github.com/alphagov/govuk-app-deployment/blob/58fcfab4262da0ba06a568dd5e9bbd76be2f4f62/lib/docker_tag_pusher.rb)], once automatic deployments are enabled for [all of the supported apps](https://github.com/alphagov/publishing-e2e-tests/blob/8412c23c5907a41a3d8b2c9dcd52d4905b139e32/docker-compose.yml). At this point, we will have significantly improved the testing for these apps, to the extent that the cost of keeping the E2E tests is far greater than any protection we might get from them.
```

Comment:
Small suggestion that just moves words around. I think this reads a bit more clearly:

```suggestion
Once automatic deployments are enabled for [all of the supported apps](https://github.com/alphagov/publishing-e2e-tests/blob/8412c23c5907a41a3d8b2c9dcd52d4905b139e32/docker-compose.yml) we will disable and delete these tests, along with all of their associated infrastructure [[1](https://github.com/alphagov/email-alert-api/blob/0b87e62288ddb1653451f84e1f36e17ce4e8e9dc/Dockerfile)] [[2](https://github.com/alphagov/govuk-app-deployment/blob/58fcfab4262da0ba06a568dd5e9bbd76be2f4f62/lib/docker_tag_pusher.rb)]. At this point, we will have significantly improved the testing for these apps, to the extent that the cost of keeping the E2E tests is far greater than any protection we might get from them.
```